apps/api/src/server/applications/application/documents/document.controller.js

```diff
@@ -140,78 +140,112 @@ export const updateDocuments = async ({ body }, response) => {
 	const { status: publishedStatus, redacted: isRedacted, documents } = body[''];
 	const formattedResponseList = [];
 
-	let redactedStatus;
-
 	// special case - this fn can be called without setting redaction status - in which case a redaction status should not be passed in to the update fn
 	// and the redaction status of each document should remain unchanged.
-	if (typeof isRedacted !== 'undefined') {
-		redactedStatus = getRedactionStatus(isRedacted);
-	}
+	const redactedStatus = isRedacted !== undefined ? getRedactionStatus(isRedacted) : undefined;
 
 	// special case - for Ready to Publish, need to check that required metadata is set on all the files - else error
-	if (publishedStatus === 'ready_to_publish') {
-		const documentIds = documents.map((/** @type {{ guid: string; }} */ document) => document.guid);
+	/** @type {{ publishableIds: string[], errors: { guid: string, msg: string }[] }} */
```

Comment:
I have a few thoughts on this endpoint overall, I think it's quite confusing to read (not that you've made it this way, it was always confusing!)

Can we separate the anonymous function to a different call entirely, only call it if the status is being set to 'ready_to_publish', and then just throw an error if there are _any_ documents we can't update? I know this breaks the contract, but it makes the whole thing read quite confusing (e.g. publishableDocuments being used throughout, and just defaulting this to all documents which seems like a workaround for our own code)

Reply:
Would it not be better to collect the errors and and return them with the successful results?

I agree about moving the anonymous function out to a named function, I'll make that change. I think it should be possible to improve the readability without sacrificing the API interface we want though.

Reply:
I've pushed a change that should make the controller more readable.